src/server/auth/arboristClient.js

```diff
@@ -21,11 +21,22 @@ class ArboristClient {
       },
     ).then(
       (response) => response.json(),
-      (err) => {
-        log.error(err);
-        throw new CodedError(500, err);
-      },
-    );
+    ).then((result) => {
+      const data = {
+        resources: [],
+      };
+      Object.keys(result).forEach((key) => {
+        if (result[key] && result[key].some((x) => x.method === 'read')) {
```

Comment:
i think you also need to handle `method: '*'`
could you include that in the test as well?

Reply:
back to this comment, for `method: '*'` should we also check for `service`?

this section

Reply:
oof, good question... right now we use "read on all services" when we grant read access, but we shouldn't assume that's the case
i guess we should check for sheepdog, peregrine or *
but then we have the same problem in windmill https://github.com/uc-cdis/data-portal/blob/2020.07/src/authMappingUtils.js#L63

Reply:
Right, so for portal now we are not checking `read` for each projects specifically. So in the submission page you will see a project name with `Browse Data` button show up even if you don't have read access to that project. But if you click that button you will see nothing (before the 404 redir) or be redirected to 404 page.

But we can make some improvements to windmill so it won't display the project in submission table if you don't have access

Reply:
mm i thought that was already the case

Reply:
can a user have only either
> {
>   "service": "sheepdog",
>   "method": "*"
> }
or
> {
>   "service": "peregrine",
>   "method": "*"
> }

Reply:
> mm i thought that was already the case

yeah you are right

Reply:
yeah i think either sheepdog or peregrine should be fine... 😕

Reply:
actually i'm starting to think we should just check "service=guppy or *"

Reply:
> actually i'm starting to think we should just check "service=guppy or *"

this actually sounds reasonable to me, even if you have access to peregrine or sheepdog doesn't mean you will be granted with access to guppy.

Reply:
right, and it won't break anything we currently do, since we just grant read access to all services